src/Umbraco.Abstractions/Services/IIdkMap.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using Umbraco.Core.Models;
+
+namespace Umbraco.Core.Services
+{
+    public interface IIdkMap
+    {
+        void SetMapper(UmbracoObjectTypes umbracoObjectType, Func<int, Guid> id2key, Func<Guid, int> key2id);
```

Comment:
This should not be included on the interface. The usage of this is strange and the way to work around it is to have a non-web based implementation of IIdkMap and a web based version of it. The non-web based version doesn't have to change, however the Web based version would have a custom implementation for `GetIdForKey` and `GetKeyForId` where it checks specifically for `UmbracoObjectTypes.Document` and `UmbracoObjectTypes.Media` and uses a different lookup mechanism for it - which would use the PublishedSnapshotService instead, then we would remove the code in PublishedSnapshotService where it calls SetMapper.

Reply:
I'm not 100% how you see the implementation, can you do this part? - The end result with a slimmer interface sounds good.

Reply:
Just pushed a change for this, will have a look at the `DistributedCacheBinder_Handlers` now

Reply:
Looks good :)